### PR TITLE
Update Go to 1.23

### PR DESCRIPTION
**Pull request by tklauser**
Cherry-picked from https://github.com/cilium/cilium/pull/34362

It seems the bot is unable to update `cilium-{builder,runtime}` images on that PR (https://github.com/cilium/cilium/actions/runs/10523063087/job/29157020472?pr=34362) so doing it manually on this PR.

Closes #34362

### CODE REVIEW

**Comment by tklauser** [comment]
Looks like renovate also doesn't update the new cilium-cli Dockerfile: https://github.com/cilium/cilium/actions/runs/10523568379/job/29158520555?pr=34542

Sent https://github.com/cilium/cilium/pull/34545 to fix.

**Comment by tklauser** [comment]
Only k8s API generation is failing now: https://github.com/cilium/cilium/actions/runs/10526157411/job/29166417076?pr=34542

I tried to investigate, but I'm confused. Running `make generate-k8s-api-local` locally with Go 1.23 works fine. But against this PR's branch it fails locally with the same error. But the PR only updates images and GH action workflows and the make target from above seemingly doesn't using any container images :thinking: 

cc @aanm @christarazi who might have more insights on the k8s API generation
cc @sayboras who also has been looking into this, it seems

**Comment by christarazi** [comment]
@tklauser Could you try this PR on top of https://github.com/cilium/cilium/pull/34463?

**Comment by sayboras** [comment]
:wave: 

yeah, it seems the changes for go directive statement in go.mod file (i.e. 1.22 -> 1.23.0) did break the k8s generation task, something wrong from upstream I think.

**Review by aanm — APPROVED** [review]
@tklauser it looks the issue is because `go.mod` has 1.23 in it. I've reverted that change alone.

**Review by kaworu — APPROVED** [review]
Hubble changes LGTM

**Review by sayboras — APPROVED** [review]
LGTM

Just curious if we need to remove dependabot config for go directive in go.mod, or just let the upgrade fail :thinking:

**Comment by aanm** [comment]
> LGTM
> 
> Just curious if we need to remove dependabot config for go directive in go.mod, or just let the upgrade fail 🤔

@sayboras I've opened https://github.com/cilium/cilium/pull/34642 because of this.

File: go.mod [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module github.com/cilium/cilium
 
 // renovate: datasource=golang-version depName=go
-go 1.22.0
```
Comment (sayboras) [inline go.mod:4]: should we revert this change as mentioned in above comment ?

**Review by youngnick — APPROVED** [review]
